core/runtime/wasm_edge/module_factory_impl.cpp

```diff
@@ -386,6 +384,15 @@ namespace kagome::runtime::wasm_edge {
       OUTCOME_TRY(writeFileTmp(path_compiled, code));
       return outcome::success();
     }
+    std::error_code ec;
+    std::filesystem::create_directories(path_compiled.parent_path(), ec);
+    if (ec) {
+      return CompilationError{fmt::format(
+          "Compiled module destination dir '{}' does not exist and cannot be "
+          "created: {}",
+          path_compiled.parent_path().string(),
+          ec.message())};
+    }
```

Comment:
[Created by `AppConfigurationImpl`](https://github.com/qdrvm/kagome/blob/master/core/application/impl/app_configuration_impl.cpp#L1476).
Also must use `mkdirs` instead of ambigous `std::filesystem::create_directories`.
```suggestion
```